### PR TITLE
Medication Statement template

### PR DESCRIPTION
**Pull request by daliboz**
Applied the new template to medication statement. This includes:
* Template changes
* Documenting authentication types
* Documenting by id queries
* Documenting custom extensions.
* Overview
* Instructions to query for "active"

### CODE REVIEW

File: content/millennium/dstu2/medications/medication-statement.md [hunk 1]
```diff
@@ -7,6 +7,39 @@ title: MedicationStatement | DSTU 2 API
...
+
+* [Patient](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.patient){:target="_blank"}
+* [Source of information](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.informationSource){:target="_blank"}
+* [Date/Time recorded](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dateAsserted){:target="_blank"}
+* [Status](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.status){:target="_blank"}
+* [Was or Was Not Taken](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.wasNotTaken){:target="_blank"}
+* [Date/Time started and ended](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.effective_x_){:target="_blank"}
+* [Medication](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.medication_x_){:target="_blank"}
+* [Category Extension](#extensions)
+* Details of medication taken:
```
Comment (jfrhong) [inline content/millennium/dstu2/medications/medication-statement.md]: We also return siteCodeableConcept and rateRatio

File: content/millennium/dstu2/medications/medication-statement.md [hunk 2]
```diff
@@ -7,6 +7,39 @@ title: MedicationStatement | DSTU 2 API
 * TOC
 {:toc}
 
+## Overview
+
+The Medication Statement Resource provides a snapshot in time of known medications taken by the patient now or in the past reported by either the patient, significant other or a provider. Future orders are not returned. Documented historical/past/home medications are commonly captured when taking the patient's medical history. Prescriptions without documented compliance are Intended, since we may not know if the patient is actively taking the medication or has filled the prescription. Medications are assumed to be Taken unless documented otherwise. 
+<br/><br/>
+References to implicitRules and modifierExtensions are NOT supported and will fail a Create or Update request.
+<br/><br/>
+The following fields are returned if valued:
```
Comment (jfrhong) [inline content/millennium/dstu2/medications/medication-statement.md]: Should we also indicate that supportingInformation will be populated with a reference to MedicationOrder?

File: content/millennium/dstu2/medications/medication-statement.md [hunk 3]
```diff
@@ -19,49 +52,127 @@ title: MedicationStatement | DSTU 2 API
...
+
+### Headers
+
+<%= headers %>
+
+### Example
+
+#### Request
+
+    GET https://fhir-open.sandboxcerner.com/dstu2/d075cf8b-3261-481d-97e5-ba6c48d3b41f/MedicationStatement?patient=4342008
```
Comment (jfrhong) [inline content/millennium/dstu2/medications/medication-statement.md]: Should this be updated to FHIRPLAY?

File: content/millennium/dstu2/medications/medication-statement.md [hunk 4]
```diff
@@ -160,12 +288,24 @@ To successfully PUT a MedicationStatement, the following headers must be provide
     x-xss-protection → 1; mode=block
 </pre>
 
-[MedicationStatement.informationSource]: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.informationSource
-[MedicationStatement.medication]: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.medication_x_
+The `ETag` response header indicates the current `If-Match` version to use on subsequent updates.
+
+### Errors
+
+The common [errors] may be returned.
```
Comment (jfrhong) [inline content/millennium/dstu2/medications/medication-statement.md]: The current errors don't list 412 which would be returned if no "if-match" header is provided

File: content/millennium/dstu2/medications/medication-statement.md [hunk 5]
```diff
@@ -7,6 +7,39 @@ title: MedicationStatement | DSTU 2 API
...
+References to implicitRules and modifierExtensions are NOT supported and will fail a Create or Update request.
+<br/><br/>
+The following fields are returned if valued:
+
+* [Patient](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.patient){:target="_blank"}
+* [Source of information](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.informationSource){:target="_blank"}
+* [Date/Time recorded](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dateAsserted){:target="_blank"}
+* [Status](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.status){:target="_blank"}
+* [Was or Was Not Taken](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.wasNotTaken){:target="_blank"}
+* [Date/Time started and ended](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.effective_x_){:target="_blank"}
```
Comment (jfrhong) [inline content/millennium/dstu2/medications/medication-statement.md]: The effective[x] isn't always the start/stop... not sure what would be a better description though

Comment (daliboz) [inline content/millennium/dstu2/medications/medication-statement.md]: There's a jira around this to try to determine if it's the mapping that's incorrect or we need to add new doc. Will wait to see how that falls out...

File: content/millennium/dstu2/medications/medication-statement.md (hunk 4 above)
Comment (daliboz) [inline content/millennium/dstu2/medications/medication-statement.md]: fixed

File: content/millennium/dstu2/medications/medication-statement.md (hunk 2 above)
Comment (daliboz) [inline content/millennium/dstu2/medications/medication-statement.md]: fixed

File: content/millennium/dstu2/medications/medication-statement.md (hunk 1 above)
Comment (daliboz) [inline content/millennium/dstu2/medications/medication-statement.md]: fixed

File: content/millennium/dstu2/medications/medication-statement.md (hunk 3 above)
Comment (daliboz) [inline content/millennium/dstu2/medications/medication-statement.md]: fixed

**Comment by daliboz** [comment]
Added updates, waiting to rebase until other +1s are acheived

**Comment by torresdrew** [comment]
+1

File: content/millennium/dstu2/medications/medication-statement.md (hunk 4 above)
Comment (torresdrew) [inline content/millennium/dstu2/medications/medication-statement.md]: +1

**Comment by jfrhong** [comment]
+1

File: content/millennium/dstu2/medications/medication-statement.md [hunk 6]
```diff
@@ -7,6 +7,42 @@ title: MedicationStatement | DSTU 2 API
...
+  * [Order comments/special instructions](http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.note){:target="_blank"}
+  * [Patient friendly display Extension](#extensions)
+  
+### Querying for Active Medications
+
+To get all possible current medications, an application should query `MedicationStatement` with the `status` query parameter set to `active,intended`. Since `MedicationStatement` is a snapshot in time, this is only a representation of what the system knew of during the last contact with the patient, and will not include things that have happened since the patient last visited with their provider.
+
+To get the list of current medications that would likely be shown by default to a practitioner, then the `MedicationOrder` resource should be used in addition to the query above in order to ensure that `draft` orders are included. Duplicates can be removed using the `MedicationStatement.supportingInformation` reference. A duplicate is identified when `MedicationOrder.id` is equivalent to the `supportingInformation` referenced `MedicationOrder/[id]`)
```
Comment (MaxPhilips) [inline content/millennium/dstu2/medications/medication-statement.md]: Extra parenthesis at the end of this line

Also I think it reads better to drop 'then' from the first sentence

**Comment by MaxPhilips** [comment]
+1 after comment

**Pull request merged**